src/main/java/org/jnode/fs/xfs/extent/DataExtent.java

```diff
@@ -34,20 +48,34 @@
      */
     private int state;
 
+    /**
+     * The msb.
+     */
+    private long msb;
+
+
     /**
      * Creates a new extent from the packed on disk format.
      *
+     * See XFS Algorithms & Data Structures
+     * Chapter 16 - page 115
+     *
+     * TODO Add structure.
+     *
      * @param data the data to read from.
      * @param offset the offset to read from.
      */
     public DataExtent(byte[] data, int offset) {
         super(data, offset);
 
-        blockCount = (int) (getUInt32(0xc) & 0x3ffffL);
-
-        startBlock = ((getUInt32(0xc) & 0xfffc0000L) >> 21) | (getUInt32(0x8) << 14) | ((getUInt32(0x4) & 0x3fL) << 38);
-
-        startOffset = ((getUInt32(0x0) & 0x7fffffff) << 22) | ((getUInt32(0x4) & 0xfffffcL) >> 9);
+        long valueUpper128bit = getInt64(0);
+        long valueLower128bit = getInt64(8);
+        blockCount = (int) (valueLower128bit & 0x1f_ff_ffL);            // 21 bits
+        valueLower128bit = valueLower128bit >>> 21;
+        startBlock = valueLower128bit | (valueUpper128bit & 0x1ffL);    // 52 bits
```

Comment:
```suggestion
        startBlock = valueLower128bit | ((valueUpper128bit & 0x1ffL) << 43);    // 52 bits
```

Reply:
The suggested change was applied,  thanks